apis/Google.Cloud.Datastore.V1/Google.Cloud.Datastore.V1/DatastoreDbImpl.cs

```diff
@@ -81,6 +81,70 @@ public DatastoreDbImpl(string projectId, string namespaceId, DatastoreClient cli
             return new LazyDatastoreQuery(streamer.Sync());
+        }
+
+        /// <inheritdoc/>
+        public override AggregationQueryResults RunAggregationQuery(AggregationQuery query, ReadConsistency? readConsistency = null, CallSettings callSettings = null)
+        {
+            var request = new RunAggregationQueryRequest
+            {
+                AggregationQuery = query,
+                ProjectId = ProjectId,
+                PartitionId = _partitionId,
+                ReadOptions = GetReadOptions(readConsistency)
+            };
+            return ProcessAggregationQueryResponse(Client.RunAggregationQuery(request, callSettings));
+        }
+
+        /// <inheritdoc/>
+        public override AggregationQueryResults RunAggregationQuery(GqlQuery query, ReadConsistency? readConsistency = null, CallSettings callSettings = null)
+        {
+            var request = new RunAggregationQueryRequest
+            {
+                GqlQuery = query,
+                ProjectId = ProjectId,
+                PartitionId = _partitionId,
+                ReadOptions = GetReadOptions(readConsistency)
+            };
+            return ProcessAggregationQueryResponse(Client.RunAggregationQuery(request, callSettings));
+        }
+
+        /// <inheritdoc/>
+        public override async Task<AggregationQueryResults> RunAggregationQueryAsync(AggregationQuery query, ReadConsistency? readConsistency = null, CallSettings callSettings = null)
+        {
+            var request = new RunAggregationQueryRequest
+            {
+                AggregationQuery = query,
+                ProjectId = ProjectId,
+                PartitionId = _partitionId,
+                ReadOptions = GetReadOptions(readConsistency)
+            };
+            var runAggregationQueryResponse = await Client.RunAggregationQueryAsync(request, callSettings).ConfigureAwait(false);
+            return ProcessAggregationQueryResponse(runAggregationQueryResponse);
+        }
+
+        /// <inheritdoc/>
+        public override async Task<AggregationQueryResults> RunAggregationQueryAsync(GqlQuery query, ReadConsistency? readConsistency = null, CallSettings callSettings = null)
+        {
+            var request = new RunAggregationQueryRequest
+            {
+                GqlQuery = query,
+                ProjectId = ProjectId,
+                PartitionId = _partitionId,
+                ReadOptions = GetReadOptions(readConsistency)
+            };
+            var runAggregationQueryResponse = await Client.RunAggregationQueryAsync(request, callSettings).ConfigureAwait(false);
+            return ProcessAggregationQueryResponse(runAggregationQueryResponse);
+        }
+
+        internal AggregationQueryResults ProcessAggregationQueryResponse(RunAggregationQueryResponse runAggregationQueryResponse)
+        {
+            var results = new AggregationQueryResults(runAggregationQueryResponse.Batch.MoreResults, runAggregationQueryResponse.Batch.ReadTime);

```

Comment:
Instead of creating the AggregationQueryResults and then modifying it, let's do everything we need to beforehand, then pass it all to the factory method / constructor. (That will mean you really don't need any internal access to `_aggregateProperties`, and don't need to create a property to expose it.)

Another alternative is to effectively move this method to AggregationQueryResults - have a factory method of:

```csharp
internal static AggregationQueryResults FromRunAggregationQueryResponse(RunAggregationQueryResponse response)
```

I think that would be nicer, personally.